File: modules/x-engine-jsi-device/iOS/Class/JSI_device.m
//
//  JSI_device.m
//  device
//
//  Created by zk on 2020/9/7.
//  Copyright © 2020 x-engine. All rights reserved.


#import "JSI_device.h"
#import "JSIContext.h"
#import "XENativeContext.h"
#import "iDevice.h"

@interface JSI_device()
@property(nonatomic,strong) id<iDevice> device;
@end

@implementation JSI_device
JSI_MODULE(JSI_device)

- (void)afterAllJSIModuleInited {
    self.device = XENP(iDevice);
}

- (NSString *)_getNavigationHeight {
    return [self.device getNavigationHeight];
}

- (NSString *)_getScreenHeight {
    return [self.device getScreenHeight];
}

- (NSString *)_getStatusBarHeight {
    return [self.device getStatusHeight];
}

- (NSString *)_getTabbarHeight {
    return [NSString stringWithFormat:@"%.2f" ,[self.device getTabbarHeight]];
}

- (NSString *)_callPhone:(phoneDto *)dto {
    return[self.device callPhone:dto.phoneNum];
}

- (NSString *)_sendMessage:(phoneDto *)dto {
    return [self.device sendMsgWithPhoneNum:dto.phoneNum withMsg:dto.phoneMsg];
}

- (void)_getDeviceInfo:(void (^)(DeviceDTO *, BOOL))completionHandler {
    NSMutableDictionary *dict = [self.device getDeviceInfo];
    DeviceDTO *dto = [[DeviceDTO alloc] init];
    dto.UUID = dict[@"UUID"];
    dto.type = dict[@"type"];
    dto.language = dict[@"language"];
    dto.systemVersion = dict[@"systemVersion"];
    completionHandler(dto, true);
}
@end

File: modules/x-engine-jsi-device/android/jsi-device/src/main/java/com/zkty/jsi/JSI_device.java
package com.zkty.jsi;

import com.alibaba.fastjson.JSON;
import com.zkty.nativ.core.NativeContext;
import com.zkty.nativ.core.NativeModule;
import com.zkty.nativ.device.IDevice;
import com.zkty.nativ.device.NativeDevice;
import com.zkty.nativ.jsi.bridge.CompletionHandler;

public class JSI_device extends xengine_jsi_device {


    private NativeDevice iDevice;

    @Override
    protected void afterAllJSIModuleInited() {
        NativeModule module = NativeContext.sharedInstance().getModuleByProtocol(IDevice.class);
        if (module instanceof NativeDevice)
            iDevice = (NativeDevice) module;

    }

    @Override
    public String _getStatusBarHeight() {
        return iDevice.getStatusBarHeight();
    }

    @Override
    public String _getNavigationHeight() {
        return iDevice.getNavigationHeight();
    }

    @Override
    public String _getScreenHeight() {
        return iDevice.getScreenHeight();
    }

    @Override
    public String _getTabbarHeight() {
        return iDevice.getTabbarHeight();
    }


    @Override
    public String _callPhone(phoneDto dto) {
        return iDevice.callPhone(dto.phoneNum);
    }

    @Override
    public String _sendMessage(phoneDto dto) {
        return iDevice.sendMessage(dto.phoneNum, dto.phoneMsg);
    }


    @Override
    public void _getDeviceInfo(CompletionHandler<DeviceDTO> handler) {
        String json = iDevice.getDeviceInfo();
        handler.complete(JSON.parseObject(json, DeviceDTO.class));
    }


}
